ricochet.pro

```diff
@@ -40,6 +40,46 @@ CONFIG += c++11
 
 VERSION = 1.1.0
 
+CONFIG(hardening):unix:!macx {
+    message("Using additional compiler flags for hardening purposes.")
+
+    SANITIZER_FLAGS = -fsanitize=address -fsanitize=undefined -fsanitize=integer-divide-by-zero -fvtable-verify=std
+
+    system(which g++-5) | system(g++ --version 2>/dev/null | grep -e "5.[0-9]") {
```

Comment:
this makes me really uneasy, and I don't think you should do it. it would break cross compilation totally, as well as any other use of a non-system-installed toolchain (which in qmake's case would be using a different mkspec)..

I'd recommend not touching QMAKE_CC/CXX, and looking into version detection of the g++ version by using system() on QMAKE_CC or something to add to the sanitize flags (which is not really portable, but slightly less broken at least)